### PR TITLE
Changed the URL for C for Python Programmers by Carl Burch 

### PR DESCRIPTION
**Pull request by Harshita-Kohli**
## What does this PR do?
Add resource(s) | Remove resource(s) | Add info | Improve repo
Improves Repo
### Description
Updated the link for C for Python Programmers by Carl Burch by removing the old link and updating it with the new link.
### Why is this valuable (or not)?
Valuable, because it helps us to directly navigate to Carl Burch's free book on C for Python Programmers without going through the broken link.
### For book lists, is it a book? For course lists, is it a course? etc.
I have made changes to the book list for the book named "C for Python Programmers" By Carl Burch.
## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/master/CONTRIBUTING.md)
- [x] Search for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction)

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: books/free-programming-books-langs.md [hunk 1]
```diff
@@ -427,7 +427,7 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 * [Beej's Guide to the GNU Debugger (GDB)](http://beej.us/guide/bggdb/) - B. Hall (HTML)
 * [Build Your Own Lisp](http://www.buildyourownlisp.com) - Daniel Holden
 * [C Elements of Style](http://www.oualline.com/books.free/style/) - Steve Oualline
-* [C for Python Programmers - Carl Burch](http://www.toves.org/books/cpy/)
+* [C for Python Programmers - Carl Burch](http://www.cburch.com/books/cpy/index.html)
```
Comment (davorpa) [inline books/free-programming-books-langs.md:430]: Maybe better this according to CONTRIBUTING url guidelines :wink:

```suggestion
* [C for Python Programmers - Carl Burch](http://www.cburch.com/books/cpy/)
```

**Review by davorpa — CHANGES_REQUESTED** [review]

File: books/free-programming-books-langs.md (hunk 1 above)
Comment (Harshita-Kohli) [reply books/free-programming-books-langs.md:430]: Sure. Will do the needful

Comment (Harshita-Kohli) [reply books/free-programming-books-langs.md:430]: Hey. I have made the changes according to contributing url guidelines

Comment (davorpa) [reply books/free-programming-books-langs.md:430]: Not reflected yet in this PR branch.

To be sure that upstream is synced, submit changes over your forked branch: `Harshita-Kohli-patch-1`

Comment (Harshita-Kohli) [reply books/free-programming-books-langs.md:430]: ok sure.

Comment (davorpa) [reply books/free-programming-books-langs.md]: Now that is submitted you can mark this conversation as resolved

**Review by davorpa — APPROVED** [review]
Resolves #6526

File: books/free-programming-books-langs.md (hunk 1 above)
Comment (Harshita-Kohli) [reply books/free-programming-books-langs.md]: Ok great! Thanks

Comment (davorpa) [reply books/free-programming-books-langs.md]: Remember add `Resolves #6500` in PR description body

Comment (Harshita-Kohli) [reply books/free-programming-books-langs.md]: Yes yes sure

**Comment by davorpa** [comment]
I hope that, with my guide, you has learned a bit more about how upstream merge / review process works :wink::hugs:

**Comment by Harshita-Kohli** [comment]
> I hope that, with my guide, you has learned a bit more about how upstream merge / review process works 😉🤗

Yes. Thank you for your guidance! Really helped me

File: books/free-programming-books-langs.md [hunk 2]
```diff
@@ -427,7 +427,7 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 * [Beej's Guide to the GNU Debugger (GDB)](http://beej.us/guide/bggdb/) - B. Hall (HTML)
 * [Build Your Own Lisp](http://www.buildyourownlisp.com) - Daniel Holden
 * [C Elements of Style](http://www.oualline.com/books.free/style/) - Steve Oualline
-* [C for Python Programmers - Carl Burch](http://www.toves.org/books/cpy/)
+* [C for Python Programmers - Carl Burch](http://www.cburch.com/books/cpy/)
```
Comment (eshellman) [inline books/free-programming-books-langs.md:430]: can we move the author attribution to the right place, too?

Comment (Harshita-Kohli) [reply books/free-programming-books-langs.md:430]: Yes. Will do that.

**Comment by Harshita-Kohli** [comment]
Changed the author name's position. Resolves #6500

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Pull request merged**